### PR TITLE
Avoid blocking background pages

### PR DESCRIPTION
**Pull request by fregante**
Fixes https://github.com/sindresorhus/notifier-for-github/issues/211

I don't like this solution, but also I can't think of something that:

- creates a Promise in the `constructor`, because it has to be available instantly
- only runs migrations on install, resolving the Promise
- resolves the promise if the `installed` event doesn't fire

As far as I know there's no "is freshly installed" boolean, but only a conditional event.

Edit: needs real world testing

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -215,7 +215,15 @@ class OptionsSync<TOptions extends Options> {
...
-			await new Promise(resolve => chrome.runtime.onInstalled.addListener(resolve));
+			// Migrations should only run onInstalled, but if that event never happens this promise still needs to proceed
+			const timeout = await Promise.race([
+				new Promise(resolve => chrome.runtime.onInstalled.addListener(resolve)),
+				new Promise(resolve => setTimeout(resolve, 500, true))
+			]);
+
+			if (timeout === true) {
+				return;
+			}
```
Comment (notlmn) [inline index.ts:226]: Why not continue running the migrations, even after the timeout?

Just curious.

Comment (fregante) [reply index.ts:226]: The point of `onInstalled` is to only run the migrations when the extension updates, not to delay it. 

If background.js is not running after an update, migrations should not run

**Comment by fregante** [comment]
If #49 (v2.0.0-0) works well for you, also try:

```
npm install webext-options-sync@2.0.0-1
```

This is a version that includes both #49 and this PR.

Since both are breaking, it'd be great to release them together

**Comment by fregante** [comment]
Ping so we can release v2

**Comment by notlmn** [comment]
Sorry @fregante, I'm currently have a hard time to test this, working full-time on a day job.

Last time when I did test on NfG, I've had other non-related ESM issues and left it for a while. Will try again the next evening.

**Comment by notlmn** [comment]
> `npm install webext-options-sync@2.0.0-1`

Can confirm that this works, the promise successfully resolves even when `installType` is not `development`.

Haven't tested out `webext-options-sync-per-domain` though.

**Pull request merged**